records/generator.go

```diff
@@ -224,6 +226,25 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 	// complete crap - refactor me
 	for _, f := range sj.Frameworks {
 		fname := labels.AsDomainFrag(f.Name)
+		tail := fname + "." + domain + "."
+		if f.WebuiURL != "" {
+			furl, _ := url.Parse(f.WebuiURL)
+			fhost, fport, _ := net.SplitHostPort(furl.Host)
+			if fhost != "" && fport != "" {
+				// if fhost is an ip add an A record
+				if net.ParseIP(fhost) != nil {
+					arec := "webui" + "." + tail
+					rg.insertRR(arec, fhost, "A")
+					fhost = arec
+				} else {
+					fhost += "."
+				}
+				tcp := "_webui" + "._tcp." + tail
```

Comment:
why _webui as the service? since the web ui URL can actually specify a protocol, why not attempt to pull the protocol from the parsed URL? if none, then does it make sense to default to _http?